cf/fetch_catalog_test.go

```diff
@@ -0,0 +1,101 @@
+package cf_test
+
+import (
+	"github.com/Peripli/service-broker-proxy-cf/cf"
+	"github.com/Peripli/service-broker-proxy/pkg/platform"
+	"github.com/cloudfoundry-community/go-cfclient"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"net/http"
+)
+
+var _ = Describe("Client FetchCatalog", func() {
+	var (
+		config         *cf.ClientConfiguration
+		client         *cf.PlatformClient
+		ccServer       *ghttp.Server
+		testBroker     *platform.ServiceBroker
+		ccResponseCode int
+		ccResponse        interface{}
+		ccResponseErr cf.CloudFoundryErr
+		expectedRequest interface{}
+		err error
+	)
+
+	BeforeEach(func() {
+		testBroker = &platform.ServiceBroker{
+			GUID:      "test-testBroker-guid",
+			Name:      "test-testBroker-name",
+			BrokerURL: "http://example.com",
+		}
+
+		ccServer = fakeCCServer(false)
+
+		config, client = ccClient(ccServer.URL())
+
+		expectedRequest = &cfclient.UpdateServiceBrokerRequest{
+			Name:      testBroker.Name,
+			BrokerURL: testBroker.BrokerURL,
+			Username:  config.Reg.User,
+			Password:  config.Reg.Password,
+		}
+
+		ccServer.AppendHandlers(
+			ghttp.CombineHandlers(
+				ghttp.VerifyRequest("PUT", "/v2/service_brokers/"+testBroker.GUID),
+				ghttp.VerifyJSONRepresenting(expectedRequest),
+				ghttp.RespondWithJSONEncodedPtr(&ccResponseCode, &ccResponse),
+			),
+		)
+	})
+
+	AfterEach(func() {
+		ccServer.Close()
+	})
+
+	Describe("Fetch", func() {
+		Context("when the call to UpdateBroker is successful", func() {
```

Comment:
Maybe assert that UpdateBroker is actually called? Lines 44-48 do say that the call is for the update-broker endpoint but the actual function call is not checked

Reply:
we can assert that the request was sent and recevied but the utils for this logic is introduced in different PR. I will add the check there.